lib/rbac/filterer.rb

```diff
@@ -254,10 +259,26 @@ def search(options = {})
 
       scope = include_references(scope, klass, include_for_find, exp_includes, skip_references)
       scope = scope.limit(limit).offset(offset) if attrs[:apply_limit_in_sql]
+
+      # putting this into an inner view will allow us to apply the virtual attributes to LIMIT rows not all of them.
+      if options[:use_sql_view]
+        if attrs[:apply_limit_in_sql] &&
+           limit && extra_cols &&
+           !klass.table_name&.include?(".") && scope.respond_to?(:includes_values)
+          inner_scope = scope.except(:select, :includes, :references)
+          scope.includes_values.each { |hash| inner_scope = add_joins(klass, inner_scope, hash) }
+
+          if !options[:skip_counts] && (attrs[:apply_limit_in_sql] && limit)
+            auth_count = inner_scope.except(:offset, :limit, :order).count(:all)
+          end
+          scope = scope.from(Arel.sql("(#{inner_scope.to_sql})").as(scope.table_name)).except(:offset, :limit, :order, :where)
```

Comment:
Pedantic (don't change):  I prefer "leading dots" for long lines like this:

```ruby
scope = scope.from(Arel.sql("(#{inner_scope.to_sql})").as(scope.table_name))
             .except(:offset, :limit, :order, :where)
```

But I don't think that `rubocop` does, so again, don't change.

Reply:
This will read better. Will see if rubocop can live with it. I read the don’t change comment but it will be nice